client.go

```diff
@@ -924,6 +924,48 @@ func (c *Client) MkdirAll(path string) error {
 	return nil
 }
 
+// RemoveAll delete files recursively in the directory and Recursively delete subdirectories.
+// An error will be returned if no file or directory with the specified path exists
+func (c *Client) RemoveAll(path string) error {
+
+	// Get the file/directory information
+	fi, err := c.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	if fi.IsDir() {
+		// Delete files recursively in the directory
+		files, err := c.ReadDir(path)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				// Recursively delete subdirectories
+				err = c.RemoveAll(path + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			} else {
+				// Delete individual files
+				err = c.Remove(path + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		// Delete the empty directory
+		return c.Remove(path)
```

Comment:
You’ve accidentally introduced a bug, where this code will no longer delete a file if pointed at a file, rather than a directory.

Let’s add a Test case for this.

The `return c.Remove(path)` should be where you have the `return nil` below.

Reply:
I am deleting file above using for condition, this will only be called to delete empty directory.
I also tested it by creating multiple files and directories and I am not getting any error. Also it is passing current Test case.
So I am bit confused

Reply:
I am bit confused here, file will be deleted by checking in for condition.

Also I have tested with several files and directory and it working fine and also passing through current test case.

Reply:
`os.RemoveAll` _can_ be called on a file, and so can this. It should remove the file.